src/swmmanywhere/post_processing.py

```diff
@@ -15,11 +15,103 @@
 import numpy as np
 import pandas as pd
 import yaml
+from swmmio import Model
 
 from swmmanywhere.filepaths import FilePaths
 from swmmanywhere.logging import logger
 
 
+def _fill_backslash_columns(df: pd.DataFrame, key: str) -> pd.DataFrame:
+    """Format the data into the swmmio columns.
+
+    Use the schema set out in defs/swmm_conversion.yml. Not all columns in `df` need to
+    be used, but all non-backslash columns in the `iwcolumns` list in
+    swmm_conversion.yml must be present.
+
+    Args:
+        df (pd.DataFrame): DataFrame to be formatted.
+        key (str): Key to look up in swmm_conversion.yml.
+
+    Returns:
+        pd.DataFrame: Formatted DataFrame
+    """
+    # Load conversion mapping from YAML file
+    with (Path(__file__).parent / "defs" / "swmm_conversion.yml").open("r") as file:
+        conversion_dict = yaml.safe_load(file)
+
+    data = {}
+    for swmmio_key, swmmanywhere_key in zip(
+        conversion_dict[key]["columns"], conversion_dict[key]["iwcolumns"]
+    ):
+        if swmmanywhere_key.startswith("/"):
+            data[swmmio_key] = swmmanywhere_key[1:]
+        else:
+            data[swmmio_key] = df[swmmanywhere_key]
+
+    return pd.DataFrame(data)
+
+
+io_registry = {}
+"""Registry for input/output functions."""
+
+
+def register_io(func):
+    """Decorator to register input/output functions."""
+
+    def wrapper(*args, **kwargs):
+        return func(*args, **kwargs)
+
+    if func.__name__ in io_registry:
+        logger.warning(f"{func.__name__} already in io register, overwriting.")
+    io_registry[func.__name__] = wrapper
+    return wrapper
+
+
+@register_io
+def apply_nodes(m: Model, addresses: FilePaths, **kw):
```

Comment:
Personally, I don't think that `m` is a descriptive enough variable name. Why not `model`? Is just 4 more characters and makes everything more readable.